zerver/lib/actions.py

```diff
@@ -2772,16 +2795,11 @@ def check_update_message(
     # 2. This is a topic-only edit for a (no topic) message, OR:
     # 3. This is a topic-only edit and you are an admin, OR:
     # 4. This is a topic-only edit and your realm allows users to edit topics.
-    if message.sender == user_profile:
-        pass
-    elif (content is None) and (
-        is_no_topic_msg
-        or user_profile.is_realm_admin
-        or user_profile.realm.allow_community_topic_editing
-    ):
-        pass
-    else:
-        raise JsonableError(_("You don't have permission to edit this message"))
+    if content is not None or topic_name is not None:
+        if not can_edit_content_or_topic(
+            message, user_profile, is_no_topic_msg, content, topic_name
+        ):
+            raise JsonableError(_("You don't have permission to edit this message"))
```

Comment:
I think it'd be better to do this as two commits, first one that just does the restructuring to add the new function, and then one that adds the outer `if content is not None or topic_name is not None` to fix the previous bug.

Reply:
Separated this to be two commits.